src/Xamarin.Android.Build.Tasks/Tasks/GetAotArguments.cs

```diff
@@ -306,7 +307,9 @@ string GetLdFlags(NdkTools ndk, AndroidTargetArch arch, int level, string toolPr
 				// Without the flag, `lld` will modify AOT-generated code in a way that the Mono runtime doesn't support. Until
 				// the runtime issue is fixed, we need to pass this flag then.
 				//
-				ldFlags.Append ("--no-relax");
+				if (!UseAndroidNdk) {
```

Comment:
I wonder if (at some point, not necessarily in this PR) we could detect the version of LLVM in the NDK and append `--no-relax` if that version is >= 14